pimd/pim_oil.c

```diff
@@ -205,8 +207,17 @@ struct channel_oil *pim_channel_oil_add(struct pim_instance *pim,
 		pim_channel_oil_change_iif(pim, c_oil, input_vif_index,
 					   name);
 		++c_oil->oil_ref_count;
-		/* channel might be present prior to upstream */
-		c_oil->up = pim_upstream_find(pim, sg);
+
+		if (!c_oil->up) {
+			/* channel might be present prior to upstream */
+			c_oil->up = pim_upstream_find(
+					pim, sg);
+			/* if the upstream entry is being anchored to an
+			 * already existing channel OIL we need to re-evaluate
+			 * the "Mute" state on AA OIFs
+			 */
+			pim_channel_update_mute(c_oil);
```

Comment:
In scenario where directly connected source on a LAN interface and we are not DR, channel oil without upstream created to drop packets further coming to CPU.
these entries still may not have upstream,. We need not call update_mute if upstream doesn't exist even now.

Reply:
I hear you. But I have seen cases where the c_oil hangs around with upstream. At this point I do not want to risk re-eval (the performance overhead should be relatively small).